WordPress/Classes/ViewRelated/Post/PostSettingsViewController+JetpackSocial.swift

```diff
@@ -3,39 +3,67 @@ extension PostSettingsViewController {
 
     @objc func showNoConnection() -> Bool {
         let isJetpackSocialEnabled = FeatureFlag.jetpackSocial.enabled
+        let isNoConnectionViewHidden = UserPersistentStoreFactory.instance().bool(forKey: Constants.hideNoConnectionViewKey)
         let blogSupportsPublicize = apost.blog.supportsPublicize()
         let blogHasNoConnections = publicizeConnections.count == 0
         let blogHasServices = availableServices().count > 0
 
         return isJetpackSocialEnabled
+        && !isNoConnectionViewHidden
         && blogSupportsPublicize
         && blogHasNoConnections
         && blogHasServices
     }
 
     @objc func createNoConnectionView() -> UIView {
         let services = availableServices()
-        let viewModel = JetpackSocialNoConnectionViewModel(
-            services: services,
-            onConnectTap: {
-                // TODO: Open the social screen
-                print("Connect tap")
-            }, onNotNowTap: {
-                // TODO: Add condition to hide the connection view after not now is tapped
-                print("Not now tap")
-            }
-        )
+        let viewModel = JetpackSocialNoConnectionViewModel(services: services,
+                                                           onConnectTap: onConnectTap(),
+                                                           onNotNowTap: onNotNowTap())
         let viewController = JetpackSocialNoConnectionView.createHostController(with: viewModel)
 
         // Returning just the view means the view controller will deallocate but we don't need a
         // reference to it. The view itself holds onto the view model.
         return viewController.view
     }
 
+    private func onConnectTap() -> () -> Void {
+        return { [weak self] in
+            guard let blog = self?.apost.blog,
+                  let controller = SharingViewController(blog: blog, delegate: self) else {
+                return
+            }
+            self?.navigationController?.pushViewController(controller, animated: true)
+        }
+    }
+
+    private func onNotNowTap() -> () -> Void {
+        return { [weak self] in
+            UserPersistentStoreFactory.instance().set(true, forKey: Constants.hideNoConnectionViewKey)
```

Comment:
hmm... should we consider scoping this preference per site instead of app-wide?

Reply:
Good question, I'll ask about it and then update this PR with whatever the conclusion is. Thread: p1687965538167589-slack-C05362VSXFU

Reply:
I added the `dotComID` to the key in this commit: https://github.com/wordpress-mobile/WordPress-iOS/pull/20965/commits/9265aee8aceb80b132ab34379aa613e5db40c43b